src/BlockTypes/ProductRatingCounter.php

```diff
@@ -0,0 +1,212 @@
+			return $content;
+		}
+
+		$post_id = $block->context['postId'];
+		$product = wc_get_product( $post_id );
+
+		if ( $product && $product->get_review_count() > 0 ) {
+			$product_reviews_count                    = $product->get_review_count();
+			$product_rating                           = $product->get_average_rating();
+			$parsed_attributes                        = $this->parse_attributes( $attributes );
+			$is_descendent_of_single_product_block    = $parsed_attributes['isDescendentOfSingleProductBlock'];
+			$is_descendent_of_single_product_template = $parsed_attributes['isDescendentOfSingleProductTemplate'];
+
+			$styles_and_classes            = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
+			$text_align_styles_and_classes = StyleAttributesUtils::get_text_align_class_and_style( $attributes );
+
+			/**
+			 * Filter the output from wc_get_rating_html.
+			 *
+			 * @param string $html   Star rating markup. Default empty string.
+			 * @param float  $rating Rating being shown.
+			 * @param int    $count  Total number of ratings.
+			 * @return string
+			 */
+			$filter_rating_html = function( $html, $rating, $count ) use ( $post_id, $product_rating, $product_reviews_count, $is_descendent_of_single_product_block, $is_descendent_of_single_product_template ) {
+				$product_permalink = get_permalink( $post_id );
+				$reviews_count     = $count;
+				$average_rating    = $rating;
+
+				if ( $product_rating ) {
+					$average_rating = $product_rating;
+				}
+
+				if ( $product_reviews_count ) {
+					$reviews_count = $product_reviews_count;
+				}
+
+				if ( 0 < $average_rating || false === $product_permalink ) {
+					/* translators: %s: rating */
+					$label                  = sprintf( __( 'Rated %s out of 5', 'woo-gutenberg-products-block' ), $average_rating );
+					$customer_reviews_count = sprintf(
+					/* translators: %s is referring to the total of reviews for a product */
+						_n(
+							'(%s customer review)',
+							'(%s customer reviews)',
+							$reviews_count,
+							'woo-gutenberg-products-block'
+						),
+						esc_html( $reviews_count )
+					);
+
+					if ( $is_descendent_of_single_product_block ) {
+						$customer_reviews_count = '<a href="' . esc_url( $product_permalink ) . '#reviews">' . $customer_reviews_count . '</a>';
+					} elseif ( $is_descendent_of_single_product_template ) {
+						$customer_reviews_count = '<a class="woocommerce-review-link" rel="nofollow" href="#reviews">' . $customer_reviews_count . '</a>';
+					}
+
+					$reviews_count_html = sprintf( '<span class="wc-block-components-product-rating__reviews_count">%1$s</span>', $customer_reviews_count );
+					$html               = sprintf(
+						'<div class="wc-block-components-product-rating__container">
+							%2$s
+						</div>
+						',
+						esc_attr( $label ),
+						$is_descendent_of_single_product_block || $is_descendent_of_single_product_template ? $reviews_count_html : ''
```

Comment:
I didn't analyze this code to the detail but there seems to be more stuff than necessary. For example:

```
$label = sprintf( __( 'Rated %s out of 5', 'woo-gutenberg-products-block' ), $average_rating );
```

used as `esc_attr( $label ),` in `$reviews_count_html`. We won't be displaying this, right?

Reply:
You are right, I removed some unnecessary stuff.